src/api/c++/z3++.h

```diff
@@ -3615,13 +3752,13 @@ namespace z3 {
             return c ? c : (Z3_context)s->ctx();
         }
 
-        struct scoped_cb {
+        struct scoped_cb final {
             user_propagator_base* p;
             scoped_cb(void* _p, Z3_solver_callback cb):p(static_cast<user_propagator_base*>(_p)) {
                 p->cb = cb;
             }
-            ~scoped_cb() { 
-                p->cb = nullptr; 
+            ~scoped_cb() {
+                if (p != nullptr) p->cb = nullptr;
```

Comment:
this change should be reverted

Reply:
Is p guaranteed never to be `nullptr`? If so then I agree, if not this is needed.

Reply:
changed to &

Reply:
Fixed by commit #54f04a5